libs/core/src/components/sage-avatar/sage-avatar.tsx

```diff
@@ -0,0 +1,124 @@
+import { Component, Host, h, Prop } from '@stencil/core';
+
+@Component({
+  tag: 'sage-avatar',
+  styleUrl: 'sage-avatar.scss',
+  shadow: true,
+})
+export class SageAvatar {
+
+  /**
+   * The alt for a custom user image.
+   * @defaultValue null
+   */
+  @Prop() alt?: string | null;
+
+  /**
+   * Determines whether the badge is visible or not.
+   * @defaultValue false
+   */
+  @Prop() badge? = false;
+
+  /**
+   * Determines whether the avatar functions as a dropdown trigger.
+   * @defaultValue false
+   */
+  @Prop() dropdown? = false;
+
+  /**
+   * The src for a custom user image.
+   * @defaultValue null
+   */
+  @Prop() image?: string | null;
+
+  /**
+   * Size of the avatar. Value can be preset or custom.
+   */
+  @Prop({ reflect: true }) size?: 
+  | 'xl' // 64px
+  | 'lg' // 56px
+  | 'md' // 40px
+  | 'sm' // 32px
+  | 'xs' // 24px
+  | string = 'lg';
+
+  /**
+   * Determines the variant of avatar. Changes appearance accordingly.
+   * @defaultValue customer
+   */
+  @Prop({ reflect: true }) variant?: 'customer' | 'admin' = 'customer';
+
+  private avatarSize() {
+    const sizes: { [key: string]: any } = {
+      xs: '24px',
+      sm: '32px',
+      md: '40px',
+      lg: '56px',
+      xl: '64px',
+    }
+
+    if (sizes[this.size]) {
+      return sizes[this.size];
+    } else {
+      return this.size
+    }
+  }
+
+  private renderAvatar = () => {
+    const style = {
+      height: this.avatarSize(),
+      width: this.avatarSize(),
+    };
+    return (
+      this.dropdown
+        ?
+        <button class="sage-avatar__button" type="button">
+          <div style={style}>
+            {this.renderIconOrImage()}
+            {this.renderBadge()}
+          </div>
+        </button>
+        :
+        <div style={style}>
+          {this.renderIconOrImage()}
+          {this.renderBadge()}
+        </div>
+    )
+  };
+
+  private renderBadge = () => (
+    this.badge
+    // Percentage is average size of icon in relation to total avatar size 
+    // of all preset sizes found in Figma. 
+    // Used to allow icons to scale to container size
+      && <sage-icon class="sage-avatar__badge" name="check-circle-filled" size="33.53%"></sage-icon> 
+  );
+
+  private renderIconOrImage = () => (
+    this.image
+      ? <img alt={this.alt} src={this.image} />
+      // Percentage is average size of icon in relation to total avatar size 
+      // of all preset sizes found in Figma. 
+      // Used to allow icons to scale to container size
+      : <sage-icon name="user-filled" size="33.53%"></sage-icon>
+  );
+
+  private classNames = () => (
+    {
+      'sage-avatar': true,
+      [`sage-avatar--has-image`]: this.image !== undefined || null, // Remove when FF supports :has selector
```

Comment:
I believe this conditional is adding the `--has-image` class at all times. If you check the canvas tab, the background is not being applied.

![Screenshot 2023-06-09 at 3 35 40 PM](https://github.com/Kajabi/sage/assets/1175111/1729941f-57d3-4352-b0ca-e2b9e1a420cb)

I refactored the classNames function slightly and it seems to work:

```
private classNames() {
    const classNames = ['sage-avatar'];

    if (this.image) {
      classNames.push('sage-avatar--has-image');
    }
    if (this.variant) {
      classNames.push('sage-avatar--' + this.variant);
    }

    return classNames.join('  ');
  }
```

There may be a better adjustment or approach, but hopefully, this can provide some direction.

Reply:
Thanks for this rec. This was a minor logical issue that I ended up pairing with Julian on.